ui-v2/src/components/flow-runs/flow-run-state-dialog/use-flow-run-state-dialog.tsx

```diff
@@ -0,0 +1,73 @@
+import type { FlowRun } from "@/api/flow-runs";
+import { useSetFlowRunState } from "@/api/flow-runs";
+import type { components } from "@/api/prefect";
+import type { RunStateFormValues } from "@/components/ui/run-state-change-dialog";
+import { StateBadge } from "@/components/ui/state-badge";
+import { RUN_STATES } from "@/utils/states";
+import { useCallback, useState } from "react";
+import { toast } from "sonner";
+
+export const useFlowRunStateDialog = () => {
+	const [open, setOpen] = useState(false);
+	const [flowRun, setFlowRun] = useState<FlowRun | null>(null);
+	const { mutateAsync: setFlowRunState } = useSetFlowRunState();
+
+	const handleSubmitChange = async (values: RunStateFormValues) => {
+		if (!flowRun?.id) {
+			return;
+		}
+
+		await setFlowRunState(
+			{
+				id: flowRun.id,
+				state: {
+					type: values.state as components["schemas"]["StateType"],
+					message: values.message || null,
+				},
+				force: true,
+			},
+			{
+				onSuccess: () => {
+					toast.success(
+						<div className="flex items-center gap-2">
+							Flow run state changed to{" "}
+							<StateBadge
+								type={values.state as components["schemas"]["StateType"]}
+								name={values.state}
+							/>
+						</div>,
+					);
+				},
+				onError: (error: Error) => {
+					const message =
+						error.message || "Unknown error while changing flow run state.";
+					toast.error(message);
+					throw error;
```

Comment:
It is probably not a good idea to throw it here because it'll crash the app if there's no error boundary. Showing the toast should be enough.